include/stl2/view/all.hpp

```diff
@@ -0,0 +1,62 @@
+// cmcstl2 - A concept-enabled C++ standard library
+//
+//  Copyright Eric Niebler 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/caseycarter/cmcstl2
+//
+#ifndef STL2_VIEW_ALL_HPP
+#define STL2_VIEW_ALL_HPP
+
+#include <stl2/detail/fwd.hpp>
+#include <stl2/detail/iterator/concepts.hpp>
+#include <stl2/detail/range/access.hpp>
+#include <stl2/detail/range/concepts.hpp>
+#include <stl2/view/iterator_range.hpp>
+
+STL2_OPEN_NAMESPACE {
+	namespace view {
+		namespace __all {
+			struct fn {
+				template <Range Rng>
+				requires View<__f<Rng>>
+				constexpr __f<Rng> operator()(Rng&& rng) const {
+					return std::forward<Rng>(rng);
+				}
+				template <Range Rng>
+				requires std::is_lvalue_reference_v<Rng> && !View<decay_t<Rng>>
+				constexpr ext::iterator_range<iterator_t<Rng>, sentinel_t<Rng>>
```

Comment:
Because of https://github.com/ericniebler/stl2/issues/501. I think `ref_view` is non-conforming (or should be).

Reply:
Do you think `c_string_view` should be non-conforming as well?

Reply:
It would be easy enough to make `c_string_view` conform to this: default-construct with begin pointing to `""`. The more general case of a `delimit_view` is considerably tricker. Hrm.